WalletWasabi/Stores/IndexStore.cs

```diff
@@ -413,7 +413,7 @@ private async Task TryCommitToFileAsync(TimeSpan throttle, CancellationToken can
 					var currentImmatureLines = ImmatureFilters.Select(x => x.ToLine()).ToArray(); // So we do not read on ImmatureFilters while removing them.
 					var matureLinesToAppend = currentImmatureLines.SkipLast(100);
 					var immatureLines = currentImmatureLines.TakeLast(100);
-					var tasks = new Task[] { MatureIndexFileManager.AppendAllLinesAsync(matureLinesToAppend), ImmatureIndexFileManager.WriteAllLinesAsync(immatureLines) };
+					var tasks = new Task[] { MatureIndexFileManager.AppendAllLinesAsync(matureLinesToAppend, cancel), ImmatureIndexFileManager.WriteAllLinesAsync(immatureLines, cancel) };
```

Comment:
This should not be cancelled. TryCommitToFileAsync operation must finish.

Reply:
Done.